modules/volume/src/processors/volumeregionmap.cpp

```diff
@@ -0,0 +1,223 @@
+ *
+ *********************************************************************************/
+
+#include <inviwo/volume/processors/volumeregionmap.h>
+#include <modules/base/algorithm/volume/volumevoronoi.h>
+
+#include <algorithm>
+#include <functional>
+#include <iostream>
+#include <fstream>
+#include <map>
+#include <unordered_map>
+
+namespace inviwo {
+
+// The Class Identifier has to be globally unique. Use a reverse DNS naming scheme
+const ProcessorInfo VolumeRegionMap::processorInfo_{
+    "org.inviwo.VolumeRegionMap",                               // Class identifier
+    "Volume Region Map",                                        // Display name
+    "Volume Operation",                                         // Category
+    CodeState::Stable,                                          // Code state
+    Tag::CPU | Tag{"Volume"} | Tag{"Atlas"} | Tag{"DataFrame"}  // Tags
+};
+const ProcessorInfo VolumeRegionMap::getProcessorInfo() const { return processorInfo_; }
+
+VolumeRegionMap::VolumeRegionMap()
+    : Processor()
+    , dataFrame_("mapping_indexes")
+    , inport_("inport")
+    , outport_("outport")
+    , from_{"from", "From", dataFrame_, ColumnOptionProperty::AddNoneOption::No, 1}
+    , to_{"to", "to", dataFrame_, ColumnOptionProperty::AddNoneOption::No, 2}
+    , missingValues_{"missingValues", "Set unfound values", 0, 0, 100} {
+
+    addPort(inport_);
+    addPort(dataFrame_);
+    addPort(outport_);
+    addProperties(from_, to_);
+    addProperties(missingValues_);
+}
+
+namespace {
+constexpr auto copyColumn = [](const Column& col, auto& dstContainer, auto assign) {
+    col.getBuffer()->getRepresentation<BufferRAM>()->dispatch<void, dispatching::filter::Scalars>(
+        [&](auto buf) {
+            const auto& data = buf->getDataContainer();
+            for (auto&& [src, dst] : util::zip(data, dstContainer)) {
+                std::invoke(assign, src, dst);
+            }
+        });
+};
+
+}
+
+void VolumeRegionMap::process() {
+    // Get columns
+    auto csv = dataFrame_.getData();
+    auto oldIdx = csv->getColumn(from_.getSelectedValue());
+    auto newIdx = csv->getColumn(to_.getSelectedValue());
+    auto nrows = csv->getNumberOfRows();
+
+    int missingValue = missingValues_.get();
+
+    // Store in vectors
+    std::vector<uint32_t> sourceIndices(nrows);
+    std::vector<uint32_t> destinationIndices(nrows);
+
+    copyColumn(*oldIdx, sourceIndices,
+               [](const auto& src, auto& dst) { dst = static_cast<uint32_t>(src); });
+    copyColumn(*newIdx, destinationIndices,
+               [](const auto& src, auto& dst) { dst = static_cast<uint32_t>(src); });
+
+    // Volume
+    auto inVolume = inport_.getData();
+    auto newVolume = std::shared_ptr<Volume>(inVolume->clone());
+    remap(newVolume, sourceIndices, destinationIndices, missingValue);
+
+    outport_.setData(newVolume);
+}
+
+void VolumeRegionMap::remap(std::shared_ptr<Volume>& volume, std::vector<unsigned int> src,
```

Comment:
Consider moving this function to a folder within the module: algorithm/volume (see for example https://github.com/inviwo/inviwo/tree/master/modules/base/include/modules/base/algorithm/volume ).

Reply:
Would be less error prone if a map is used instead of two separate vectors. Or perhaps there is a particular reason for using two vectors as input?

Reply:
The two vectors originate from two separate DataFrame columns. Using a map with the corresponding preconditions would only offload that functionality to the caller.